### PR TITLE
CDT-2: added filterProductProperties step and modified getProductProp…

### PR DESCRIPTION
**Pull request by aaron-martin**
Added the property filtering step

### CODE REVIEW

File: pipelines/getProductProperties_v1.json [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+          }
+        ]
+      },
+      {
+        "id": "@shopgate/minimum-advertising-pricing",
+        "path": "@shopgate/minimum-advertising-pricing/filterProductProperties.js",
+        "type": "extension",
+        "input": [
+          {
+            "id": "105",{
```
Comment (devbucket) [inline pipelines/getProductProperties_v1.json]: I think this curly bracket doesn't belong here ;)

**Review by devbucket — CHANGES_REQUESTED** [review]

File: pipelines/getProductProperties_v1.json (hunk 1 above)
Comment (aaron-martin) [reply pipelines/getProductProperties_v1.json]: corrected stupid copy paste mishap

File: extension/filterProductProperties.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = function (context, input, cb) {
+  const propertiesToRemove = ['Apply Map']
+  const propertiesToReturen = input.properties.filter(function (property) {
```
Comment (pvomhoff) [inline extension/filterProductProperties.js]: There is a typo in `propertiesToReturen`

**Review by pvomhoff — CHANGES_REQUESTED** [review]

File: extension/filterProductProperties.js (hunk 2 above)
Comment (aaron-martin) [reply extension/filterProductProperties.js]: corrected name

File: extension/flagMapProducts.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+module.exports = function (context, input, cb) {
+  const triggerProps = [
+    {
+      label: 'Apply Map',
+      value: 'yes'
+    }
+  ]
+  for (let product of input.products) {
```
Comment (pvomhoff) [inline extension/flagMapProducts.js]: You can do this much easier by using:
[`Array.filter`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and [`Array.map`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

File: extension/adjustProductsForMapPricing.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = function (context, input, cb) {
+  for (let product of input.products) {
+    if (product.flags.isMapProduct) {
```
Comment (pvomhoff) [inline extension/adjustProductsForMapPricing.js]: You also has to set the `msrp` price to null

File: extension/flagMapProducts.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+module.exports = function (context, input, cb) {
+  const triggerProps = [
+    {
+      label: 'Apply Map',
```
Comment (pvomhoff) [inline extension/flagMapProducts.js:4]: The `label` of the apply map have a different capatalization like: `Apply MAP` same goes for `value`.

You can make sure that they fitting by just make the label and the value lowercase by a function and then compare it

**Review by pvomhoff — CHANGES_REQUESTED** [review]

File: extension/adjustProductsForMapPricing.js (hunk 4 above)
Comment (aaron-martin) [reply extension/adjustProductsForMapPricing.js]: added line to set msrp line to null when isMapProduct flag is set to true.

File: extension/flagMapProducts.js (hunk 5 above)
Comment (aaron-martin) [reply extension/flagMapProducts.js:4]: made comparison of map label and value so it is case insensitive.

File: extension/flagMapProducts.js (hunk 3 above)
Comment (aaron-martin) [reply extension/flagMapProducts.js]: removed for loop and replaced it with the use of the map method of the Array class.

File: extension/adjustProductsForMapPricing.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+module.exports = function (context, input, cb) {
+  const adjustedProducts = input.products.map((product) => {
+    if (product.flags.isMapProduct) {
+      product.price.unitPrice = null
```
Comment (pvomhoff) [inline extension/adjustProductsForMapPricing.js]: We just looked it up in the theme:
We are currently checking if the price object is not null we dont show it.
So we can do `product.price = null`

File: extension/filterProductProperties.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+module.exports = function (context, input, cb) {
```
Comment (pvomhoff) [inline extension/filterProductProperties.js:1]: This extension it self should be used for any shop so white list are atm very customer specific.
We shouldnt put it in this extension.
I would just filter out the `Apply Map` property.

File: extension/flagMapProducts.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+/**
+ * @param properties
+ * @param triggerProps
+ * @return {boolean}
+ */
+function hasTriggerProperty (properties, triggerProps) {
+  for (let property of properties) {
+    let foundObject = triggerProps.find(triggerProp => triggerProp.label.toLowerCase() === property.label.toLowerCase() && triggerProp.value.toLowerCase() === property.value.toLowerCase())
```
Comment (pvomhoff) [inline extension/flagMapProducts.js]: Please split this check off to multiple lines.
Its better readable

**Review by pvomhoff — CHANGES_REQUESTED** [review]

File: extension/filterProductProperties.js (hunk 7 above)
Comment (aaron-martin) [reply extension/filterProductProperties.js:1]: The functionality of this script now only filters out the apply map property.

File: extension/flagMapProducts.js (hunk 8 above)
Comment (aaron-martin) [reply extension/flagMapProducts.js]: I split this into multiple lines.

**Review by pvomhoff — APPROVED** [review]

**Pull request merged**